frontend/src/app/main/data/workspace/comments.cljs

```diff
@@ -95,8 +104,76 @@
         (rx/merge
          (rx/of (rt/nav :workspace pparams qparams))
          (->> stream
-              (rx/filter (ptk/type? ::dw/initialize-viewport))
+              (rx/filter (ptk/type? ::dwv/initialize-viewport))
               (rx/take 1)
               (rx/mapcat #(rx/of (center-to-comment-thread thread)
-                                 (dw/select-for-drawing :comments)
+                                 (dwd/select-for-drawing :comments)
                                  (dcm/open-thread thread)))))))))
+
+(defn update-comment-thread-position
+  ([thread  [new-x new-y]]
+   (update-comment-thread-position thread  [new-x new-y] nil))
+
+  ([thread  [new-x new-y] frame-id]
+  (us/assert ::dcm/comment-thread thread)
+  (ptk/reify ::update-comment-thread-position
+    ptk/WatchEvent
+    (watch [it state _]
+      (let [thread-id (:id thread)
+            page (wsh/lookup-page state)
+            page-id (:id page)
+            objects (wsh/lookup-page-objects state page-id)
+            new-frame-id (if (nil? frame-id)
+                           (cph/frame-id-by-position objects {:x new-x :y new-y})
+                           (:frame-id thread))
+            thread (assoc thread
+                          :position {:x new-x :y new-y}
+                          :frame-id new-frame-id)
+
+            changes
+            (-> (pcb/empty-changes it)
+                (pcb/with-page page)
+                (pcb/update-page-option :comment-threads-position assoc thread-id (select-keys thread [:position :frame-id])))]
+
+        (rx/merge
+         (rx/of (dwc/commit-changes changes))
+         (->> (rp/mutation :update-comment-thread-position thread)
+              (rx/catch #(rx/throw {:type :update-comment-thread-position}))
+              (rx/ignore))))))))
+
+(defn move-frame-comment-threads
+  "Move comment threads that are inside a frame when that frame is moved"
+  [ids]
+  (us/verify (s/coll-of uuid?) ids)
+
+  (ptk/reify ::move-frame-comment-threads
+    ptk/WatchEvent
+    (watch [_ state _]
+      (let [objects (wsh/lookup-page-objects state)
+
+            is-frame? (fn [id] (= :frame (get-in objects [id :type])))
+            frame-ids? (into #{} (filter is-frame?) ids)
+
+            object-modifiers  (get state :workspace-modifiers)
```

Comment:
if the keyword is known at compile time, just use it as function instead of `get`